sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -828,21 +829,28 @@ private async Task RunReceiveTaskAsync(CancellationToken cancellationToken)
                             break;
                         }
 
-                        // Do a quick synchronous check before we resort to async/await with the state-machine overhead.
-                        if (!_messageHandlerSemaphore.Wait(0, CancellationToken.None))
+                        bool messageHandlerLockAcquired = false;
```

Comment:
Why remove the quick sync check?

Reply:
It was mostly just for code readability - now that we have two semaphores that we are locking it becomes messy to handle the sync check. I will look into adding this back for both semaphores.